graph/patterns/change-status.md

```diff
@@ -0,0 +1,107 @@
+
+Taking in consideration the example of the `riskyUser` above, it is possible to keep the `riskState` property on the resource:
+
+```xml
+<EnumType Name="riskState">
+    <Member Name="confirmedCompromised" Value="0"/>
+    <Member Name="dismissed" Value="1"/>
+    <Member Name="none" Value="2"/>
+</EnumType>
+
+<EntityType Name="riskyUser" >
+    <Property Name="riskState" Type="riskState" />
+</EntityType>
+```
+
+To confirm the user as compromised, we can send a `PATCH` request on the resource:
+
+```bash
+curl -X PATCH --json '{"riskState": "confirmedCompromised"}' https://graph.microsoft.com/v1.0/identityProtection/riskyUsers/a57dc75f-24b5-47ce-b5e1-44822f5d4729
+```
+
+Instead of returning an empty body with `204` status code, depending on the situation, the response should contain the status of the request along with other properties modified because of the request, if it can be satisfied in a reasonable time:
+
+```json
+{
+  "riskState": "confirmedCompromised",
+  "riskDetail": "adminConfirmedUserCompromised",
+  "riskLastUpdatedDateTime": "2015-06-19T12-01-03.4Z",
+  "desiredRiskState": "confirmedCompromised"
+}
+```
+
+If the response triggers a long running operation, the response should contain a reference to query its status, following the [existing guidance][2]:
+
+```json
+{
+  "createdDateTime": "2015-06-19T12-01-03.4Z",
+  "riskState": "confirmingCompromising"
+}
+```
+
+The status on the resource shall stay the same until the long running operation is completed, but the workload can optionally return a property representing the desired status along with the current resource status to signal that it is working on a request:
+
+curl https://graph.microsoft.com/v1.0/identityProtection/riskyUsers/a57dc75f-24b5-47ce-b5e1-44822f5d4729
+
+```json
+{ "riskState": "none", "desiredRiskState": "confirmedCompromised" } // Other properties omitted for brevity
+```
+
+---
+
+There might be cases where the change of the status might require some additional parameters, or where we might need to execute an action that is not 100% related to the entity. In such case, a `ChangeStatusRequest` resource might be created to accommodate this use case.
+
+For instance, let's say the `riskyUser` confirmation process also requires a parameter that defines for how much time the user is going to be disabled. In this case, instead of a `riskyState` property on the exposed resource, we can define a new resource and create a new instance representing the intention:
+
+```xml
+<EnumType Name="riskState">
+    <Member Name="confirmedCompromised" Value="0"/>
+    <Member Name="dismissed" Value="1"/>
+    <Member Name="unknown" Value="2"/>
+</EnumType>
+
+<EntityType Name="riskAssessment" >
+    <Property Name="riskState" Type="riskState" />
+    <Property Name="duration" Type="Edm.Duration" />
```

Comment:
I'm not sure what to do with duration property. For richLongRunningOperation we have percentageComplete. Does duration serve a similar role?

Reply:
This duration property has nothing to do with the Long Running Operations; it is an hypothetical example on how a `riskState` could be help up to that time. The general idea is to show how to model situation where the status change requires an additional parameter.

Does that explain?